include/ck/tensor_operation/gpu/grid/gridwise_gemm_reduce_xdl_cshuffle_v1.hpp

```diff
@@ -852,22 +856,27 @@ struct GridwiseGemmReduce_k0mk1_k0nk1_mn_xdl_cshuffle_v1
                     // reduce in VGPR
                     static_for<0, mreduce_per_thread, 1>{}([&](auto im) {
                         FloatReduceAcc d0_acc = d0_reduce_op.GetReduceZeroValue();
-                        FloatReduceAcc d1_acc = d1_reduce_op.GetReduceZeroValue();
+                        FloatReduceAcc d1_acc = 0;
+                        if(enable_d1)
+                            d1_acc = d1_reduce_op.GetReduceZeroValue();
 
                         static_for<0, nreduce_per_thread, 1>{}([&](auto in) {
                             constexpr auto offset =
                                 Number<c_reduce_thread_desc_mperblock_nperblock.CalculateOffset(
                                     make_tuple(im, in))>{};
 
                             d0_reduce_op.Reduce(d0_acc, c_reduce_thread_buf[offset]);
-                            d1_reduce_op.Reduce(d1_acc, c_reduce_thread_buf[offset]);
+
+                            if(enable_d1)
+                                d1_reduce_op.Reduce(d1_acc, c_reduce_thread_buf[offset]);
```

Comment:
I fear that putting conditional statements inside the loop can have perf impacts. My take is as long as it doesn't involve writing back to memory, and reduce op for d1 typically computes nothing when d1 is `nullptr`, you can remove the `if()` without any perf impact.

Reply:
fix